plugins/powerd/fu-plugin-powerd.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2021 Twain Byrnes <binarynewts@google.com>
+ * Copyright (C) 2021 George Popoola <gpopoola@google.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#define MINIMUM_BATTERY_PERCENTAGE_FALLBACK 10
+
+struct FuPluginData {
+	GDBusProxy *proxy; /* nullable */
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	if (data->proxy != NULL)
+		g_object_unref (data->proxy);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GDBusConnection) connection = NULL;
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	/* Establishing DBus connection and proxy for method call to powerd */
+	connection = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
+
+	if (connection == NULL) {
+		g_prefix_error (error,
+				"Connection with system was not established: ");
+		return FALSE;
+	}
+
+	data->proxy = g_dbus_proxy_new_sync (connection,
+					     G_DBUS_PROXY_FLAGS_NONE,
+					     NULL,
+					     "org.chromium.PowerManager",
+					     "/org/chromium/PowerManager",
+					     "org.chromium.PowerManager",
+					     NULL,
+					     error);
+
+	if (data->proxy == NULL) {
+		g_prefix_error (
+		    error,
+		    "Failed to establish proxy for power manager server: ");
+		return FALSE;
+	}
+
```

Comment:
As we discussed on the call this morning, you may need to do a test method call so that systems that don't have powerd correctly disable themselves.

See how fu-plugin-upower.c does it here:
https://github.com/fwupd/fwupd/blob/master/plugins/upower/fu-plugin-upower.c#L100